scripts/xdist/pytest_container_manager.py

```diff
@@ -0,0 +1,187 @@
+import argparse
+import logging
+import time
+
+import boto3
+from botocore.exceptions import ClientError
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+
+class PytestContainerManager():
+    """
+    Responsible for spinning up and terminating containers to be used with pytest-xdist
+    """
+
+    def __init__(self, region, cluster):
+        self.ecs = boto3.client('ecs', region)
+        self.cluster_name = cluster
+
+    def spin_up_containers(self, number_of_containers, task_name, subnets, security_groups, public_ip_enabled, launch_type):
+        """
+        Spins up containers and generates two .txt files, one containing the IP
+        addresses of the new containers, the other containing their task_arns.
+        """
+        CONTAINER_RUN_TIME_OUT_MINUTES = 10
+        MAX_RUN_TASK_RETRIES = 7
+
```

Comment:
For a long method that is more procedural as opposed to computational, it would be helpful to have some additional comments concerning what is going on

Reply:
done